src/mainwindow.cpp

```diff
@@ -36,22 +36,29 @@ MainWindow::MainWindow(QWidget *parent) :
     ui->baudRate->setCurrentIndex(3);
     loadPortsAndSet();
 
-    connect(this, &MainWindow::startMonitor, m_monitor, &Monitor::readAndParse);
-    m_monitor->moveToThread(&m_monitorThread);
-    m_monitorThread.start();
-
-    connect(ui->monitorButton, &QToolButton::toggled, this, &MainWindow::handleMonitorToggle);
+    connect(ui->monitorButton, &QToolButton::toggled, this, &MainWindow::handleMonitorToggled);
     connect(ui->clearButton, &QToolButton::released, ui->plainTextEdit, &QPlainTextEdit::clear);
     connect(ui->sendButton, &QToolButton::released, this, &MainWindow::handleSend);
     connect(ui->portReload, &QToolButton::released, this, &MainWindow::handleReloadPorts);
     connect(ui->port, static_cast<void (QComboBox::*)(int index)>(&QComboBox::currentIndexChanged), this, &MainWindow::handlePortChanged);
-    connect(this, &MainWindow::open, m_monitor, &Monitor::open);
     connect(ui->baudRate, static_cast<void (QComboBox::*)(int index)>(&QComboBox::currentIndexChanged), this, &MainWindow::handleBaudRateChanged);
-    connect(m_monitor, &Monitor::output, this, &MainWindow::handleOutput);
-    connect(m_monitor, &Monitor::error, this, &MainWindow::handleError);
     connect(ui->lineEdit, &QLineEdit::returnPressed, this, &MainWindow::handleSend);
-    connect(ui->plotterButton, &QToolButton::toggled, this, &MainWindow::handlePlotter);
-
+    connect(ui->plotterButton, &QToolButton::toggled, this, &MainWindow::handlePlotterToggled);
+    connect(&m_serialPort, &QSerialPort::errorOccurred, this, &MainWindow::handleError);
+
+    connect(ui->plainTextEdit->verticalScrollBar(), &QScrollBar::sliderPressed, [this]() {
+        m_monitorVerticalScrollBarGrabbing = true;
+    });
+    connect(ui->plainTextEdit->verticalScrollBar(), &QScrollBar::sliderReleased, [this]() {
+        m_monitorVerticalScrollBarGrabbing = false;
+    });
+
+    worker = new Worker;
+    worker->moveToThread(&workerThread);
+    workerThread.start();
+    connect(this, &MainWindow::sendToWorker, worker, &Worker::processData);
+
+// Apply bundled icons on Windows and macOS
 #if defined _WIN32 || defined __APPLE__
```

Comment:
Since you're using Qt I'm pretty sure Qt defines macros for OS detection, forget what they are though

Reply:
I just found a better way to do this. Instead of checking for os, I can set the theme icon in the code rather than the UI form, and QIcon lets you specify a fallback icon, which would be the bundled icons in this case. I'll change this